### ISSUE TITLE
chunk classification using longformer?

### ISSUE THREAD

**Issue by vr25** [allenai/longformer#123]
Hi,

Any plans on classification script by pooling chunks of a long document? (similar to [triviaqa.py](https://github.com/allenai/longformer/blob/master/scripts/triviaqa.py))

Thanks!

**Comment by armancohan** [comment]
We already have a script for classification now but it currently lives in the [classification branch](https://github.com/allenai/longformer/tree/classification)
Please check out this script: https://github.com/allenai/longformer/blob/classification/scripts/classification.py

**Comment by vr25** [comment]
Thanks for sharing it, yes, I already checked it out before opening this issue. However, I couldn't find the chunking pooling logic implemented in this script. Something like this in triviaqa.py https://github.com/allenai/longformer/blob/master/scripts/triviaqa.py#L349

Please let me know if I am missing something.

Thanks!

**Comment by vr25** [comment]
@ibeltagy wrt my previous question [here](https://github.com/allenai/longformer/issues/62#issuecomment-646117367)

Did you fine-tune RoBERTa (unfreeze all 12 layers) for creating document vectors using chunking-pooling logic?

Below is what I did for BERT:
1. Break the long document into 512 token sequences and pad the start and end with [CLS] and [SEP] tokens respectively.
2. So now each document is a list of a variable number of token sublists.
3. Then I pass a document list into the forward layer (BERT params: requires_grad=True) such that for each sublist I obtain a CLS head vector. So if I have 120 chunks, I will get 120 such vectors which are then pooled (say, using mean()) to create a final document vector.

**Issue:** Since I have set requires_grad=True, for each document chunk, a separate copy BERT gets created. So this document will create 120 such copies and it runs into CUDA OOM error. I thought of the data and model parallelism (opened an issue [here](https://discuss.pytorch.org/t/calculate-gradients-for-bert-fine-tune-on-multiple-gpus/98111) but no response yet).

Any quick solutions to fix this problem?

Thanks!
